common/lib/dependabot/pull_request_creator/message_builder/link_and_mention_sanitizer.rb

```diff
@@ -61,35 +63,43 @@ def sanitize_mentions(text)
 
             last_match = Regexp.last_match
 
-            sanitized_mention = mention.gsub("@", "@&#8203;")
             if last_match.pre_match.chars.last == "[" &&
                last_match.post_match.chars.first == "]"
-              sanitized_mention
+              next mention
```

Comment:
We should probably keep this as `sanitized_mention` until we use commonmarker here too. This code is assuming the mention is inside a link `[@mention](url)` but it could also be a mention in square brackets: [@feelepxyz] (which notifies me)

Reply:
Also would be ace with a test for this case!